explorer/src/pages/TransactionDetailsPage.tsx

```diff
@@ -410,6 +415,129 @@ function AccountsCard({
   );
 }
 
+type TokenBalanceRow = {
+  account: PublicKey;
+  mint: string;
+  balance: TokenAmount;
+  delta: number;
+  accountIndex: number;
+};
+
+function generateTokenBalanceRows(
+  preTokenBalances: TokenBalance[],
+  postTokenBalances: TokenBalance[],
+  accounts: ParsedMessageAccount[]
+): TokenBalanceRow[] {
+  let preBalanceMap: { [index: number]: TokenBalance } = {};
+
+  preTokenBalances.forEach(
+    (balance) => (preBalanceMap[balance.accountIndex] = balance)
+  );
+
+  return postTokenBalances
+    .map(
+      ({ uiTokenAmount, accountIndex, mint }): TokenBalanceRow => {
+        let delta;
+
+        if (accountIndex in preBalanceMap) {
+          delta = normalizeTokenDelta(
+            uiTokenAmount,
+            preBalanceMap[accountIndex].uiTokenAmount
+          );
```

Comment:
Yes, I started out doing that but there was some strange JS precision stuff going on, which was adding decimals. I would normally use math.js in that situation, but this way didn't involve that dependency. 

![Explorer-Solana (3)](https://user-images.githubusercontent.com/188792/106081362-9fc54580-60cd-11eb-8cc7-9edbbbf7c0b9.png)

Reply:
It would be nice to have a math library that properly handles precision actually.

Looks like [mathjs is built on decimal.js](https://mathjs.org/docs/datatypes/bignumbers.html)

Maybe we should go with bignumber.js? Oyster lending has been using bn.js but I don't think it's as nice for displaying precise values as strings.

https://dev.to/fvictorio/a-comparison-of-bignumber-libraries-in-javascript-2gc5
> bignumber.js and decimal.js are similar, the main difference is that bignumber.js expresses its precision in terms of decimals (appropriate for financial applications, for example) and decimal.js does it in terms of significant digits (better for scientific applications). That's why decimal.js is not a good choice for arbitrary integer arithmetic1.

Reply:
Cool, I'll give bignumber a try.